### PR TITLE
Add optional GL wrapper that checks for errors after each API call.

### LINKED ISSUE #166: Add mode that asserts when glGetError returns non-zero after any GL call
Making this configurable at build time would make it much nicer to debug mysterious errors.

### PR DESCRIPTION
**Pull request by jdm**
Fixes #166.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/gleam/167)

### CODE REVIEW

File: src/gl.rs [hunk 1]
```diff
@@ -518,6 +518,447 @@ pub trait Gl {
...
+            let rv = self.gl.$name($($arg,)*);
+            assert_eq!(self.gl.get_error(), 0);
+            rv
+        })+
+    }
+}
+
+impl<T: Gl> Gl for ErrorCheckingGl<T> {
+    gl_apis! {
+        fn get_type(&self) -> GlType;
```
Comment (kvark) [inline src/gl.rs]: Is it possible to re-use this list of signatures for the `trait Gl` to avoid repeating ourselves?

Comment (jdm) [reply src/gl.rs]: I extended the macro to take the entire trait declaration as input.

File: src/gl_fns.rs [hunk 2]
```diff
@@ -15,9 +15,10 @@ impl GlFns
 {
     pub unsafe fn load_with<'a, F>(loadfn: F) -> Rc<Gl> where F: FnMut(&str) -> *const c_void {
         let ffi_gl_ = GlFfi::load_with(loadfn);
-        Rc::new(GlFns {
+        let fns = GlFns {
             ffi_gl_: ffi_gl_,
-        }) as Rc<Gl>
+        };
+        ErrorCheckingGl::maybe_wrap(fns)
```
Comment (kvark) [inline src/gl_fns.rs]: What's most bothering me about the PR is that the switch to use asserting wrapper can really be a run-time switch (as opposed to compile time feature). Can't we just expose `ErrorCheckingGl` to the user and let them call `wrap` if they want to?

Comment (jdm) [reply src/gl_fns.rs]: That's fair.

**Comment by nox** [comment]
I don't see a solution to handle both unsafe and not-unsafe methods, and this is blocking me in #164.

File: src/gl.rs [hunk 3]
```diff
@@ -293,229 +298,252 @@ pub trait Gl {
...
-        &self,
-        program: GLuint,
-        name: &str,
-    ) -> GLint;
+        fn alias_point_size_range(&self) -> (GLfloat, GLfloat);
+        fn alias_line_width_range(&self) -> (GLfloat, GLfloat);
 
-    fn alias_point_size_range(&self) -> (GLfloat, GLfloat);
-    fn alias_line_width_range(&self) -> (GLfloat, GLfloat);
+        // Returns the the maximum supported width and height of the viewport.
```
Comment (nox) [inline src/gl.rs]: I also disagree with this patch if it leads to docblocks being removed, though I kill this method anyway in my own patch.

**Pull request merged**